service/src/test/java/bio/terra/workspace/common/utils/WorkspaceActivityLogHookTest.java

```diff
@@ -0,0 +1,169 @@
+package bio.terra.workspace.common.utils;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+
+import bio.terra.common.exception.MissingRequiredFieldException;
+import bio.terra.stairway.FlightDebugInfo;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.db.WorkspaceActivityLogDao;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.job.JobServiceTestFlight;
+import bio.terra.workspace.service.workspace.model.OperationType;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import bio.terra.workspace.service.workspace.model.WorkspaceStage;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+public class WorkspaceActivityLogHookTest extends BaseUnitTest {
+  private static final AuthenticatedUserRequest testUser =
+      new AuthenticatedUserRequest()
+          .subjectId("StairwayUnit")
+          .email("stairway@unit.com")
+          .token(Optional.of("not-a-real-token"));
+
+  @Autowired private JobService jobService;
+  @Autowired private WorkspaceDao workspaceDao;
+  @Autowired private WorkspaceActivityLogDao activityLogDao;
+  @MockBean private SamService mockSamService;
+
+  @BeforeEach
```

Comment:
Delete 41-49. I'm deleting these in JobServiceTest in #694.

45 is throwing an exception because the mock isn't set up correctly. The exception message has details on how it's incorrect.